### DISCUSSION TITLE
Weapon attachment bonuses are misleading due to increasing weapon volume lowering the weapons performance

### DISCUSSION THREAD

**Discussion by RaynaWasTaken** [cataclysmbn/Cataclysm-BN#2049] [General]
# Is your feature request related to a problem? Please describe.
Most weapon attachments in the game are useless or actively detrimental to the player.
They are for the most part still using their original values for buffs/debuffs.
This would be fine, but at some point weapon volume increase started to affect weapon performance in a detrimental way, causing them to be slower to aim with for example.

Those debuffs eat up any advantages the weapon mods offer in many cases, oftentimes even causing weapon attachments to be actively detrimental to the weapons performance when used.

# Describe the solution you'd like
Remove the detriments to weapon stats introduced by the volume gain of the weapon when a mod is attached, it serves no real purpose as attachments can already be given negative stats directly, without having to tiptoe around some obscure blackbox mechanic in the background affecting stats further.

# Describe alternatives you've considered
Fix all weapon mods manually, either by changing their stats so they take the volume debuffs into consideration, or setting their internal volume to 0 so they work as intended.
This however would carry the problem that modded attachments will not be affected by that change and not every modder may be aware of that mechanic to begin with

**Comment by Coolthulhu** [comment]
High volume doesn't reduce aim speed by a flat or scaled amount, but rather adds a cap on aim speed. This makes it rather hard to understand without reading the source code for it.
Rifles and launchers are the most affected due to their large start volume.

Here's the exact math:
The cap is `ceil(49-volume/75ml)`, while aim speed is calculated as `2+dex+sight+skill-enc/10`, with pistols getting 2x skill and rifles getting 0.9 skill. Then the cap is reduced by 5 for rifles and increased to 125% for pistols. Then there's low cap depending on skill, but it only matters for huge guns.
Then the value is multiplied by flat 6.5 and by a non-linear multiplier between 0 and 1, depending on recoil.
Then there's low cap of 5 aim speed.

A really good character, with 12 dex, 10 skill, underbarrel laser sight, for a total of 33 aim speed (43 with pistols, 32 with rifles), would need a 900ml rifle, a 525ml pistol, or a 1.2L non-pistol-non-rifle gun to avoid the cap.
A bad character, with 8 dex, 0 skill, default sights (6 speed), for a total of 16 aim speed, would need a 2.1L rifle or a 2.475 non-pistol-non-rifle to avoid the cap.
Unless my math is wrong, this would mean that guns like AK-47, pneumatic assault rifle, or M4A1 with a magazine inserted, are aiming at cap dictated by volume rather than one by sights. Quick test in the game shows that my math was most likely correct.

Now that I did the math, it looks like it never really worked and just hid that fact well enough. This is what happens when important features aren't backed up by hard math on in-game examples. Also when players are unaware of what is happening under the hood because UI doesn't say enough.
I see the following options:
* Change the cap to a flat penalty - needs design, but not much of it since what we have doesn't work, so it's easy to improve it
* Change the cap to a divisor on speed - would have the advantage of buffing pistols more, but would hit big rifles pretty hard (maybe not a problem)
* Combine flat+multiplier penalties somehow - I'd need good "data points" to test it with
* Add aim speed penalty/divisor as a json field, but default to a formula including one of the above

**Comment by mkikt4743** [comment]
Could also add more properties display to gunmods. Right now I think it only shows handling bonus, I'd hope it could show all the affected stats

**Comment by calrissian-tendra** [comment]
If you google marine battle rifle, you'll see the M27 Infantry Automatic Rifle. Look at the attachments on it, and then think about how quickly a skilled operator can acquire targets with such a weapon. For an even more extreme example, check out the Navy Seal's M4A1, with all the attachments that might be needed for an operation, and think about how fast a skilled user could acquire targets with it.
My suggestion here is that weapon volume should have basically nothing to do with accuracy. Rather, the weight, length of weapon, and the user's skill with that class of weapon should be the only determinig factors.

**Comment by scarf005** [comment]
fixed by #3820
